src/exporter/DaemonMetricCollector.cc

```diff
@@ -80,7 +81,34 @@ std::string boost_string_to_std(boost::json::string js) {
 
 std::string quote(std::string value) { return "\"" + value + "\""; }
 
-bool is_hyphen(char ch) { return ch == '-'; }
+void promethize(std::string &name) {
+  auto should_be_underscore = [](char ch) {
+    return ch == '.' || ch == '/' || ch == ' ';
+  };
+  std::replace_if(name.begin(), name.end(), should_be_underscore, '_');
+
+  size_t double_colon_pos = name.find("::");
+  if (double_colon_pos != std::string::npos) {
+    name.replace(double_colon_pos, 2, "_");
+  }
+
+  // just in case because there are some bluestore metrics which include a single ":"
+  if (name.find(":") != std::string::npos) {
+    boost::replace_all(name, ":", "_");
```

Comment:
as the comment says, I saw a single colon around some bluestore metrics so I added that in just in case.

Reply:
Then `promethize` in `src/pybind/mgr/prometheus/module.py` needs to updated to the same.  There can't be "just in case" here, both methods should do exactly the same thing.

Reply:
Yup, it's important to keep consistency with the existing metric names (even if the current approach was not the best possible). Otherwise, this will generate new time-series, and the existing Grafana Dashboards might have missing data.

Maybe those bluestore metrics are not exported due to their priority level?